File: dotnet/src/fit/PrimitiveFixture.cs
// Modified or written by Object Mentor, Inc. for inclusion with FitNesse.
// Copyright (c) 2002 Cunningham & Cunningham, Inc.
// Released under the terms of the GNU General Public License version 2 or later.
// Copyright (c) 2002 Cunningham & Cunningham, Inc.
// Released under the terms of the GNU General Public License version 2 or later.

namespace fit
{
	public class PrimitiveFixture : Fixture
	{
		// format converters ////////////////////////
		public static long ParseLong(Parse cell) 
		{
			return long.Parse(cell.Text);
		}

		public static double ParseDouble (Parse cell) 
		{
			return double.Parse(cell.Text);
		}

		// answer comparisons ///////////////////////

		public virtual void Check(Parse cell, string actual) 
		{
			if (cell.Text == actual) 
				Right(cell);
			else 
				Wrong(cell, actual);
		}

		public virtual void Check(Parse cell, long actual) 
		{
			if (ParseLong(cell) == actual) 
				Right(cell);
			else 
				Wrong(cell, actual.ToString());
		}

		public virtual void Check(Parse cell, double actual) 
		{
			if (ParseDouble(cell) == actual) 
				Right(cell);
			else 
				Wrong(cell, actual.ToString());
		}
	}
}

File: src/main/java/fit/PrimitiveFixture.java
// Modified or written by Object Mentor, Inc. for inclusion with FitNesse.
// Copyright (c) 2002 Cunningham & Cunningham, Inc.
// Released under the terms of the GNU General Public License version 2 or later.
package fit;

// Copyright (c) 2002 Cunningham & Cunningham, Inc.
// Released under the terms of the GNU General Public License version 2 or later.


public class PrimitiveFixture extends Fixture {

  // format converters ////////////////////////

  public static long parseLong(Parse cell) {
    return Long.parseLong(cell.text());
  }

  public static double parseDouble(Parse cell) {
    return Double.parseDouble(cell.text());
  }

  // answer comparisons ///////////////////////

  public void check(Parse cell, String value) {
    if (cell.text().equals(value)) {
      right(cell);
    } else {
      wrong(cell, value);
    }
  }

  public void check(Parse cell, long value) {
    if (parseLong(cell) == value) {
      right(cell);
    } else {
      wrong(cell, Long.toString(value));
    }
  }

  public void check(Parse cell, double value) {
    if (parseDouble(cell) == value) {
      right(cell);
    } else {
      wrong(cell, Double.toString(value));
    }
  }

}
